var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1250,12 +1250,26 @@ def libs(self):
         # The values of LDLIBRARY and LIBRARY aren't reliable. Intel Python uses a
         # static binary but installs shared libraries, so sysconfig reports
         # libpythonX.Y.a but only libpythonX.Y.so exists. So we add our own paths, too.
-        shared_libs = [
-            self.config_vars["LDLIBRARY"],
+
+        # When using homebrew on macOS, self.config_vars["LDLIBRARY"] points to
+        # Python.framework/Versions/X.Y/Python, which is indeed a shared library
+        # but doesn't have the correct filename suffix. This confuses several
+        # packages (e.g. met) that use python.libs.ld_flags. Ignore this config.
```

Comment:
I think this comment should be worded more directly based on the error listed in https://github.com/spack/spack/issues/33409: we want to avoid including a directory with a library that is not linkable on Mac OS. Also, `LDLIBRARY` yields `'Python.framework/Versions/3.9/Python'` for me on my Mac OS Python and it wasn't built with Homebrew, so I think the context can be simplified to "Python on Mac OS" vs. "Python with Homebrew on Mac OS".

Reply:
It's prob best described as "Framework Python". Spack builds Python with the traditional Unix filesystem hierarchy, but Homebrew and other PMs can build framework Python instead.